examples/lakers-nrf52840/src/bin/initiator.rs

```diff
@@ -86,24 +86,37 @@ async fn main(spawner: Spawner) {
 
     match rcvd {
         Ok(pckt_2) => {
+            info!("Received message_2");
             let message_2: EdhocMessageBuffer =
                 pckt_2.pdu[1..pckt_2.len].try_into().expect("wrong length");
             let (initiator, c_r, id_cred_r, ead_2) = initiator.parse_message_2(&message_2).unwrap();
             let valid_cred_r = credential_check_or_fetch(Some(cred_r), id_cred_r).unwrap();
             let initiator = initiator.verify_message_2(valid_cred_r).unwrap();
 
-            let (mut initiator, message_3, i_prk_out) = initiator
+            let (mut initiator, message_3, i_prk_out, i_prk_out_exporter) = initiator
                 .prepare_message_3(CredentialTransfer::ByReference, &None)
                 .unwrap();
-
-            common::transmit_without_response(
-                &mut radio,
+            let pckt_3 =
                 common::Packet::new_from_slice(message_3.as_slice(), Some(c_r.as_slice()[0]))
-                    .unwrap(),
-            )
-            .await;
-
-            info!("Handshake completed. prk_out = {:X}", i_prk_out);
+                    .expect("Buffer not long enough");
+            info!("Send message_3 and wait message_4");
+            let rcvd =
+                common::transmit_and_wait_response(&mut radio, pckt_3, Some(c_r.as_slice()[0]))
+                    .await;
+
+            info!("Sent message_3");
+            match rcvd {
+                Ok(pckt_4) => {
+                    info!("Received message_4");
+                    let message_4: EdhocMessageBuffer =
+                        pckt_4.pdu[1..pckt_4.len].try_into().expect("wrong length");
```

Comment:
Why does it start at index 1?
Please add a short comment explaining.

---

As a general rule, we write comments when the code is non-obvious.
For example, there's no need to write comments like 'call tx function', 'call prepare message 1', etc.
But, if the code is like an optimization for a specific case that happens sometimes, a comment helps future readers of the code to understand it.
Or, if the code is setting configurations that are highly contenxt-dependent, and especially non-obvious to a future reader, then we should add a brief comment explaining it.

See for example [this code](https://github.com/openswarm-eu/DotBot-firmware/blob/main/bsp/nrf/clock.c#L58) from Alex, where he explains what `NRF_CLOCK->HFCLKCTRL = CLOCK_HFCLKCTRL_HCLK_Div1 << CLOCK_HFCLKCTRL_HCLK_Pos;` is doing.
Imagine figuring that out without the `// Enable 128MHZ core clock, only possible with application core` comment right above.